extensions/interactions/MathExpressionInput/MathExpressionInput.js

```diff
@@ -30,13 +30,125 @@ oppia.directive('oppiaInteractiveMathExpressionInput', [
+
+          /**
+           * Adds a button overlay and invisible text field used to bring up
+           * the keyboard on mobile devices.
+           *
+           * TODO: On iOS/Safari, keyboard may only appear on first press.
+           * This may not be a significant issue as the MathExpressionInput is
+           * recreated if the given answer is incorrect.
+           */
+          var addMobileButtonOverlay = function() {
+            /**
+             * Checks if the guppy div has a width and height greater than 0,
+             * if not schedules a timeout to run again after 100ms. If the
+             * guppy div has a valid width/height, position the invisible
+             * button directly over the guppy div. For mobile browsers, focus()
+             * can only be called from within an onclick handler. Using a form
+             * element was more reliable than attaching the handler to the
+             * guppy div directly.
+             */
+            var positionButtonOverlay = function() {
+              var guppyOffset = $(guppyDivElt).position();
+              var guppySize = document.querySelector('.guppy-div')
+                .getBoundingClientRect();
+              console.log(guppySize, $('#startMathInputButton'));
+
+              // If the guppy div hasn't rendered yet, retry after 100ms.
+              if (guppySize.width === 0 || guppySize.height === 0) {
+                $timeout(positionButtonOverlay, 100);
+              } else {
+                $('#startMathInputButton').css({
+                  top: guppyOffset.top,
+                  left: guppyOffset.left,
+                  width: guppySize.width,
+                  height: guppySize.height
+                });
+              }
+            };
+            positionButtonOverlay();
+
+            // The focus() call must be in a click event handler and on a text
+            // field to make the mobile keyboard appear.
+            $('#startMathInputButton').click(function() {
+              guppyInstance.activate();
+
+              var fakeInputElement = document.
+                querySelector('#fakeInputForMathExpression');
+              fakeInputElement.focus();
+
+              // Place the cursor at the end of the text input, so that the
+              // user can use backspace to delete.
+              fakeInputElement.setSelectionRange(fakeInputElement.value.length,
+                fakeInputElement.value.length);
+            });
+
+            // Mapping characters back to Mousetrap codes.
+            // Use Mousetrap.trigger(code) to replay them.
+            var K_SYM_REVERSE_MAP = {
+              '^': 'shift+6',
+              '*': 'shift+8',
+              '(': 'shift+9',
+              '<': 'shift+,',
+              '>': 'shift+.',
+              '\\': 'shift+\\',
+              ' ': 'space',
+              ')': 'shift+0'
+              // Missing up/down. user needs to use [space], 'sub' instead.
```

Comment:
Ah! That is a _really_ good question. I don't know the answer. We can still merge this PR and file two issues to handle the question you ask in your last paragraph in a separate pass.

IMO, or the first question, I think the most intuitive thing would be to touch the relevant area on the screen. I don't know if Guppy has support for receiving such events, but it would be nice for it to have them, I think.

For the second question, I have no idea. I'm not _too_ worried, since I don't think subscripts are going to hit us for a while, but I don't know what the correct behavior should be.

In any case, could you also please file an issue (or two) [here](https://github.com/daniel3735928559/guppy/), and ask? Thank you for bringing this up!